File: Aerolinea/src/java/airline/model/Viaje.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package airline.model;
import java.io.Serializable;
import java.util.Date;
/**
 *
 * @author AndreyCh
 */
public class Viaje implements Serializable, Jsonable{
    int numero_viaje;
    Date fecha;
    Avion avion;
    Vuelo vuelo;
    int horaLlegada;

    public Viaje(int numero_viaje, Date fecha, Avion avion, Vuelo vuelo, int horaLlegada) {
        this.numero_viaje = numero_viaje;
        this.fecha = fecha;
        this.avion = avion;
        this.vuelo = vuelo;
        this.horaLlegada = horaLlegada;
    }

    public Viaje() {
    }

    public int getNumero_viaje() {
        return numero_viaje;
    }

    public void setNumero_viaje(int numero_viaje) {
        this.numero_viaje = numero_viaje;
    }

    public Date getFecha() {
        return fecha;
    }

    public void setFecha(Date fecha) {
        this.fecha = fecha;
    }

    public Avion getAvion() {
        return avion;
    }

    public void setAvion(Avion avion) {
        this.avion = avion;
    }

    public Vuelo getVuelo() {
        return vuelo;
    }

    public void setVuelo(Vuelo vuelo) {
        this.vuelo = vuelo;
    }

    public int getHoraLlegada() {
        return horaLlegada;
    }

    public void setHoraLlegada(int horaLlegada) {
        this.horaLlegada = horaLlegada;
    }

   
}

File: Aerolinea/web/js/Viaje.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
function Viaje(numero,fecha,avion,vuelo){
	this.Viaje(numero,fecha,avion,vuelo);
}
Viaje.prototype={
        numero: 0,
        fecha:"",
        avion:"",
        vuelo: 0,
	Viaje:function(numero,fecha,avion,vuelo){
		this.numero=numero;
                this.fecha=fecha;
		this.avion=avion;
		this.vuelo=vuelo;
	},
        toString:function(){
	  return this.numero;
	}
}
Viaje.from=function(plain){
    viaje = new Viaje(plain.numero,plain.fecha,plain.avion,plain.vuelo);
    return viaje;
};

Viaje.to= function(viaje){
    return{
         _class : 'Viaje',
         numero: viaje.numero,
         fecha: viaje.fecha,
         avion: viaje.avion,
         vuelo: viaje.vuelo
    };
};



